### PR TITLE
Feature/e commerce protos Jordi

### PR DESCRIPTION
**Pull request by JordiRoronoa**

### CODE REVIEW

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/**
+ * Abstraction of Cart
+ * 
+ * @method (add)
```
Comment (maryammdot) [inline staff/jordi-pujol/e-commerce-protos/scripts/cart.js:4]: you should add the other methods to the documentation

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        if (!(item instanceof Product)) throw TypeError ('Should be an object');
+
+        this.cartItems.push(item)
+        return this.cartItems.length
+    };
+
+    Cart.prototype.totalPrice = function(){
+        var sumP = 0;
+        for (let i = 0; i< this.cartItems.length; i++){
```
Comment (maryammdot) [inline staff/jordi-pujol/e-commerce-protos/scripts/cart.js]: You could try to use the method reduce from Arrays

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+           sumP += this.cartItems[i].price
+        };
+        return sumP
+    };
+
+    Cart.prototype.numberOfItems = function (){
```
Comment (maryammdot) [inline staff/jordi-pujol/e-commerce-protos/scripts/cart.js:39]: You have chosen the complicated path! jejeje what you have done is not the same as returning the length?

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        return sumItems
+    }
+
+    Cart.prototype.mostExpensive = function() {
+        var exp = this.cartItems[0].price;
+        var pro = {}
+        for (let i=0; i< this.cartItems.length; i++){
+            if (exp < this.cartItems[i].price) {
+                exp = this.cartItems[i].price
+                pro = this.cartItems[i].constructor.name
```
Comment (maryammdot) [inline staff/jordi-pujol/e-commerce-protos/scripts/cart.js:50]: Are you sure that with the constructor.name you get the name of the product?

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.test.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            
+            var res = cart.add(socks)
+
+            var expected = 5;
+
+            expect (res === expected,'expected and result should be the same')
+        });
+    })
+
+    describe ('fails adding', function() {
```
Comment (maryammdot) [inline staff/jordi-pujol/e-commerce-protos/scripts/cart.test.js]: I would not make a new describe for the fail cases. I would put them in new it but inside the same describe than the succeed ones

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.test.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        it ('should add an item that is not an instance of product & when no param is passed', function (){
+            
+            var error;
+
+            try {
+                cart.add(5)
+            } catch (err){
+                error = err
+            }
+            expect (error,'expected and result should be the same')
```
Comment (maryammdot) [inline staff/jordi-pujol/e-commerce-protos/scripts/cart.test.js:40]: You can also check the type of error

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.test.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    });
+
+        // .totalPrice()
+
+    describe('testing cart.totalPrice()', function(){
+
+        it ('should calculate total price ', function(){
+
+            var res = cart.totalPrice()
+            var expected = 102.91999999999999
```
Comment (maryammdot) [inline staff/jordi-pujol/e-commerce-protos/scripts/cart.test.js]: you could control how many decimals you want! maybe there are too many 🤓

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.js (hunk 3 above)
Comment (JordiRoronoa) [reply staff/jordi-pujol/e-commerce-protos/scripts/cart.js:39]: Thanks, much better this way...!

File: staff/jordi-pujol/e-commerce-protos/scripts/cart.js (hunk 4 above)
Comment (JordiRoronoa) [reply staff/jordi-pujol/e-commerce-protos/scripts/cart.js:50]: Well, you are right I get the name of the constructor, but not the product one...

**Pull request merged**